### DISCUSSION TITLE
Focus / devices with iOS

### DISCUSSION THREAD

**Discussion by mynona** [alpinejs/alpine#4360] [7. Bugs (archived)]
When I set the focus with:

´´´
$nextTick(()=>{$refs.searchInput.focus()});
´´´

Everything works as expected on the desktop (Mac OS, Windows)

But on devices running the latest iOS versions (17.x) the focus is not set to the search input.

Steps to reproduce:
Run the code on a device that is on iOS

Is this the expected behavior that it should not work on iOS or is this a bug?

**Comment by ekwoka** [comment]
If it's a bug, it would 99% be a bug in Safari.

To clarify, Safari 17.5 on ios is impacted, but not Safari 17.5 on MacOS?

**Reply by m-y-n-o-n-a** [reply]
> If it's a bug, it would 99% be a bug in Safari.
> 
> 
> 
> To clarify, Safari 17.5 on ios is impacted, but not Safari 17.5 on MacOS?

yes, only the iOS version is impacted.

**Comment by SimoTod** [comment]
Likely iOS on mobile has some optimisation that delays the page repainting so 1 tick is not enough. Try setting a short timeout instead.

**Comment by m-y-n-o-n-a** [comment]
> Likely iOS on mobile has some optimisation that delays the page repainting so 1 tick is not enough. Try setting a short timeout instead.

thx, will try that

**Comment by mynona** [comment]
Unfortunately, this didn't resolve the issue:

```
$nextTick(()=>{setTimeout(()=>{$refs.searchInput.focus();},50);});
```

**Comment by ekwoka** [comment]
Oh, I think I remember something about this before.

You may need to do `el.click(); el.focus()` on iOS.

That it just doesn't like directly focusing programmatically.

**Reply by mynona** [reply]
Thank you for your answer.

Question: how can I reference it with el.focus() ?

There is a clear button next to the input. If a user wants to clear the input he would press this button and the focus is set to the input.

Works for MacOS + Windows, but not on mobile devices with iOS unfortunately.

**Reply by ekwoka** [reply]
`el` was just a filler for "whatever element you're working with"

**Comment by mynona** [comment]
@ekwoka Unfortunately, this makes no difference:

```
$nextTick(()=>{
                               setTimeout(()=>{
                                 $refs.searchInput.click();
                                 $refs.searchInput.focus();
                               },10);
                             });
```

**Reply by ekwoka** [reply]
You shouldn't need to do the setTimeout. Doesn't really make sense in that context. Could even be the thing causing the issue.

**Comment by mynona** [comment]
Nope. Nothing changed. Before I had the setTimeout inside the nextTick


```

$nextTick(()=>{
                        $refs.searchInput.click();
                        $refs.searchInput.focus();
                    });
```

**Reply by SimoTod** [reply]
As it doesn't appear to be Alpine related, you may have better look if you try to google something like "ios focus programmatically not working". For example, try https://stackoverflow.com/questions/30752250/ios-workaround-for-manually-focusing-on-an-input-textarea#30977546
